### PR TITLE
SWIFT-373 Add runCommand operation and switch DB methods to use it

### PR DESCRIPTION
**Pull request by kmahar**
This adds a `RunCommandOperation` and switches `runCommand`, `createCollection`, and `dropDatabase` to use it.

### CODE REVIEW

File: Sources/MongoSwift/Operations/RunCommandOperation.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        self.writeConcern = writeConcern
+    }
+
+    private enum CodingKeys: String, CodingKey {
+        case readConcern, writeConcern
+    }
+}
+
+/// An operation corresponding to a `runCommand` call.
+internal struct RunCommandOperation: Operation {
```
Comment (mbroadst) [inline Sources/MongoSwift/Operations/RunCommandOperation.swift]: I think we might be jumping the gun a little bit creating the `RunCommandOperation` at this stage, it brings with it many considerations that I think greatly expand the current scope of work. For starters, it's specific to a `MongoDatabase` when it really should be something we can reuse for `MongoClient`/`MongoCollection`/`MongoDatabase`, anything with a namespace really. Secondly, it appears that the mongoc command you chose will [skip database read preference inheritance](https://github.com/mongodb/mongo-c-driver/blob/master/src/libmongoc/src/mongoc/mongoc-database.c#L327).

Basically, I think this work moves us into command construction earlier than we want to. Did something necessitate the change, or was it just annoying duplicating code?

Comment (kmahar) [reply Sources/MongoSwift/Operations/RunCommandOperation.swift]: ok, maybe `RunDatabaseCommandOperation` is a better name for this? it's just copy paste from our existing `MongoDatabase.runCommand` method.
so the read preference thing is an existing bug then - we should check if the `MongoDatabase` has a read preference set and send that along if the user doesn't provide one in the options.

As for rewriting the other methods to use this, yeah, the motivation was just that it was a lot of boilerplate to create new operations and they were simple to rewrite this way. so I was trying to do something similar to the "bulk write everything" refactor.

Comment (mbroadst) [reply Sources/MongoSwift/Operations/RunCommandOperation.swift]: I think it's probably fine to keep this one named `RunCommandOperation` for now, and use it _only_ for `MongoDatabase.runCommand`. I'm worried in the other cases where you are reusing it here that we are subtly introducing bugs like I pointed to above. As you move into client pooling I think it may be much easier to see patterns for modifying `RunCommandOperation` to support use across all types.

Comment (kmahar) [reply Sources/MongoSwift/Operations/RunCommandOperation.swift]: I ended up noticing a separate unrelated ReadPreference issue so I just opened a separate PR to fix both of those. Will move dropDatabase and createCollection to use their own operations containing their original logic.

**Comment by kmahar** [comment]
Ok I've updated so that `RunCommandOperation` is only used for `MongoDatabase.runCommand`. The other DB methods have their own operation types.
Also noticed that I missed adding an operation for dropCollection so I threw that one in too.
